src/components/document/document.tsx

```diff
@@ -228,14 +234,24 @@ export class DocumentComponent extends BaseComponent<IProps, IState> {
 
   private renderOtherDocumentTitleBar(type: string, hideButtons?: boolean) {
     const {document} = this.props;
-    const { user: { isTeacher } } = this.stores;
+    const { user: { isTeacher }, documents, user } = this.stores;
+    let countNotDeleted = 0;
+    const getOtherDocument = documents.byTypeForUser((PersonalDocument || LearningLogDocument), user.id);
+    getOtherDocument.forEach(doc => {
+      const isDeletedStatus = doc.getProperty("isDeleted");
+      if (isDeletedStatus !== "true") {
+        ++countNotDeleted;
+      }
+    });
     return (
       <div className={`titlebar ${type}`}>
         {!hideButtons &&
           <div className="actions">
             <NewButton onClick={this.handleNewDocumentClick} />
             <CopyButton onClick={this.handleCopyDocumentClick} />
-            <DeleteButton onClick={this.handleDeleteDocumentClick} />
+            {countNotDeleted > 1
+              ? <DeleteButton onClick={this.handleDeleteDocumentClick} /> : <DeleteDisabledButton />
+            }
```

Comment:
```suggestion
             <DeleteButton enabled={countNotDeleted > 1} onClick={this.handleDeleteDocumentClick} />
```
As mentioned above, make the enabled/disabled state a property/behavior of the `DeleteButton` component rather than creating a separate component.